File: Runtime/TransitionAttributeTypeCollection.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Imports System.Collections.ObjectModel

''' <summary>
''' Collection of Transition Attribute Types
''' </summary>
Friend Class TransitionAttributeTypeCollection
    Inherits KeyedCollection(Of Integer, TransitionAttributeType)

    Protected Overrides Function GetKeyForItem(ByVal item As TransitionAttributeType) As Integer
        Return item.TransitionAttributeId
    End Function

End Class

File: src/Runtime/TransitionAttributeTypeCollection.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2019 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Collections.ObjectModel;

namespace SyncroSim.STSim
{
    internal class TransitionAttributeTypeCollection : KeyedCollection<int, TransitionAttributeType>
    {
        protected override int GetKeyForItem(TransitionAttributeType item)
        {
            return item.TransitionAttributeId;
        }
    }
}
